src/cmd/dev.go

```diff
@@ -104,20 +105,30 @@ var devTransformGitLinksCmd = &cobra.Command{
 	Aliases: []string{"p"},
 	Short:   lang.CmdDevPatchGitShort,
 	Args:    cobra.ExactArgs(2),
-	RunE: func(_ *cobra.Command, args []string) error {
+	RunE: func(cmd *cobra.Command, args []string) error {
 		host, fileName := args[0], args[1]
+		ctx := cmd.Context()
 
 		// Read the contents of the given file
 		content, err := os.ReadFile(fileName)
 		if err != nil {
 			return fmt.Errorf("unable to read the file %s: %w", fileName, err)
 		}
 
-		pkgConfig.InitOpts.GitServer.Address = host
+		gitServer := pkgConfig.InitOpts.GitServer
+		gitServer.Address = host
 
 		// Perform git url transformation via regex
 		text := string(content)
-		processedText := transform.MutateGitURLsInText(message.Warnf, pkgConfig.InitOpts.GitServer.Address, text, pkgConfig.InitOpts.GitServer.PushUsername)
+
+		// Set log func for transform
+		var logFn func(string, ...any)
+		logFn = message.Warnf
+		// Use logger if we can
+		if logger.Enabled(ctx) {
+			logFn = logger.From(ctx).Warn
+		}
+		processedText := transform.MutateGitURLsInText(logFn, gitServer.Address, text, gitServer.PushUsername)
```

Comment:
This is actually expecting printf style logging so the message ends up looking like this. 
```bash
 Unable to transform the git url, using the original url we have: %s !BADKEY=https://*/*.git
```
You could instead use this to keep with the printf format
```go
		l := logger.From(ctx)
		logFn = slog.NewLogLogger(l.Handler(), slog.LevelWarn).Printf
```
Might be good to add a todo to this function to move to just use the slogger from context after logging is released also. Not sure why this was given it's own logging function. Doesn't looked to be called from anywhere outside of Zarf

Reply:
Yeah that's bugged for sure -- I'm going to tear it out and make a todo for later. I can see the effort to make a logger you can pass in, but it just creates weird compatibility issues. the Printf on log.Logger is neat though, hadn't seen that before.